tests/components/jewish_calendar/conftest.py

```diff
@@ -26,3 +39,78 @@ def mock_setup_entry() -> Generator[AsyncMock]:
         "homeassistant.components.jewish_calendar.async_setup_entry", return_value=True
     ) as mock_setup_entry:
         yield mock_setup_entry
+
+
+@pytest.fixture
+def jcal_params(request: pytest.FixtureRequest) -> dict | None:
+    """Return test parameters."""
+    if not hasattr(request, "param"):
+        return None
+
+    if len(request.param) == 3:
+        location_name, test_time, results = request.param
+        havdalah_offset = 0
+
+    if len(request.param) == 4:
+        location_name, test_time, results, havdalah_offset = request.param
+
+    time_zone, latlng, candle_light = LOCATIONS[location_name]
+    tz_info = dt_util.get_time_zone(time_zone)
+    test_time = test_time.replace(tzinfo=tz_info)
+    if isinstance(results, dict):
+        results = {
+            key: value.replace(tzinfo=tz_info)
+            if isinstance(value, dt.datetime)
+            else value
+            for key, value in results.items()
+        }
+
+    return {
+        "test_time": test_time,
+        "results": results,
+        CONF_TIME_ZONE: time_zone,
+        CONF_DIASPORA: location_name not in ("Jerusalem",),
+        CONF_LOCATION: latlng,
+        CONF_CANDLE_LIGHT_MINUTES: candle_light,
+        CONF_HAVDALAH_OFFSET_MINUTES: havdalah_offset,
+    }
```

Comment:
Consider using a dataclass for this, instead of a dict

Reply:
Done